packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx

```diff
@@ -1,427 +1,187 @@
-                { item.name }
-              </span>
-            ) }
-            { hasSubmenuItems(item) ? (
-              <Icon className={ css('ms-CommandBarItem-chevronDown', styles.itemChevronDown) } iconName='ChevronDown' />
-            ) : (null) }
-          </button>;
-        } else if (item.href) {
-          return <a
-            { ...getNativeProps(item, anchorProperties) }
-            id={ this._id + item.key }
-            className={ className }
-            href={ item.href }
-            data-command-key={ index }
-            aria-haspopup={ hasSubmenuItems(item) }
-            role='menuitem'
-            aria-label={ item.ariaLabel }
-          >
-            { (hasIcon) ? this._renderIcon(item) : (null) }
-            { (!!item.name) && (
-              <span
-                className={ css('ms-CommandBarItem-commandText', styles.itemCommandText) }
-              >
-                { item.name }
-              </span>
-            ) }
-          </a>;
-        } else {
-          return <div
-            { ...getNativeProps(item, divProperties) }
-            id={ this._id + item.key }
-            className={ className }
-            data-command-key={ index }
-            aria-haspopup={ hasSubmenuItems(item) }
-          >
-            { (hasIcon) ? this._renderIcon(item) : (null) }
-            { (!!item.name) && (
-              <span
-                className={ css('ms-CommandBarItem-commandText', styles.itemCommandText) }
-                aria-hidden='true'
-                role='presentation'
-              >
-                { item.name }
-              </span>
-            ) }
-          </div>;
-        }
-      })() }
-    </div>;
+  private computeCacheKey(primaryItems: ICommandBarItemProps[]): string {
+    return primaryItems.reduce((acc, current) => acc + current.key, '');
   }
 
-  private _renderIcon(item: IContextualMenuItem) {
-    // Only present to allow continued use of item.icon which is deprecated.
-    let iconProps: IIconProps = item.iconProps ? item.iconProps : {
-      iconName: item.icon as IconName
-    };
-    // Use the default icon color for the known icon names
-    let iconColorClassName = iconProps.iconName === 'None' ? '' : ('ms-CommandBarItem-iconColor ' + styles.itemIconColor);
-    let iconClassName = css('ms-CommandBarItem-icon', styles.itemIcon, iconColorClassName, iconProps.className);
+  @autobind
+  private _onReduceData(data: ICommandBarData): ICommandBarData | undefined {
+    let { primaryItems, overflowItems, cacheKey } = data;
+    let movedItem = primaryItems[primaryItems.length - 1];
 
-    return <Icon { ...iconProps } className={ iconClassName } />;
-  }
+    if (movedItem !== undefined) {
+      movedItem.renderedInOverflow = true;
 
-  private _asyncMeasure() {
-    this._async.requestAnimationFrame(() => {
-      this._updateItemMeasurements();
-      this._updateRenderedItems();
-    });
-  }
+      overflowItems = overflowItems.concat(movedItem);
```

Comment:
I personally prefer using array spread over concat, not sure what the fabric convention is